sdk/metric/aggregator/counter/counter.go

```diff
@@ -19,54 +19,48 @@ import (
 
 	"go.opentelemetry.io/otel/api/core"
 	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregator"
 )
 
 // Aggregator aggregates counter events.
 type Aggregator struct {
 	// current holds current increments to this counter record
 	current core.Number
 
-	// checkpoint is a temporary used during Collect()
+	// checkpoint is a temporary used during Checkpoint()
 	checkpoint core.Number
 }
 
 var _ export.Aggregator = &Aggregator{}
+var _ aggregator.Sum = &Aggregator{}
 
 // New returns a new counter aggregator.  This aggregator computes an
 // atomic sum.
 func New() *Aggregator {
 	return &Aggregator{}
 }
 
-// AsNumber returns the accumulated count as an int64.
-func (c *Aggregator) AsNumber() core.Number {
-	return c.checkpoint.AsNumber()
+// Sum returns the accumulated count as a Number.
+func (c *Aggregator) Sum() core.Number {
+	return c.checkpoint
 }
 
-// Collect checkpoints the current value (atomically) and exports it.
-func (c *Aggregator) Collect(ctx context.Context, rec export.Record, exp export.Batcher) {
+// Checkpoint checkpoints the current value (atomically) and exports it.
```

Comment:
Maybe rephrase it to 
```
Checkpoint atomically checkpoints the current value and exports it.
```

Reply:
also needs a comment stating that it will, in fact reset the counter back to 0 for the next period's start.